server/plugin/oAuth.go

```diff
@@ -149,21 +195,25 @@ func (p *Plugin) GenerateOAuthToken(code, state string) error {
 		return err
 	}
 
+	tokenExpiryDurationInSeconds, err := strconv.Atoi(successResponse.ExpiresIn)
+	if err != nil {
+		if _, DMErr := p.DM(mattermostUserID, constants.GenericErrorMessage); DMErr != nil {
+			return DMErr
+		}
+		return err
+	}
+
 	user := store.User{
 		MattermostUserID: mattermostUserID,
 		AccessToken:      p.encode(encryptedAccessToken),
 		RefreshToken:     p.encode(encryptedRefreshToken),
-		ExpiresIn:        successResponse.ExpiresIn,
+		ExpiresAt:        time.Now().UTC().Add(time.Second * time.Duration(tokenExpiryDurationInSeconds)).Unix(),
 	}
 
 	if err := p.Store.StoreUser(&user); err != nil {
 		return err
 	}
 
-	if _, err := p.DM(mattermostUserID, fmt.Sprintf("%s\n\n%s", constants.UserConnected, constants.HelpText)); err != nil {
-		return err
-	}
-
 	p.API.PublishWebSocketEvent(
```

Comment:
Why is `GenerateAndStoreOAuthToken` function publishing the websocket event here. It's job should be only what the name specifies i.e. generate and store the oauth token.

Reply:
we need to publish websocket event for account connection when the token is successfully generated and stored so, placed that code here after generate and store process

are you suggesting to create a separate func for this webscoket event?

Reply:
below comment makes sense